### DISCUSSION TITLE
Add device classes to `sensor` for CO and CO2

### DISCUSSION THREAD

**Discussion by iMicknl** [home-assistant/architecture#520] [Entity Models] [answered]
Currently there is no `DEVICE_CLASS` for CO and CO2, while HomeKit is requiring this device class for the right mapping. They have been defined in `homekit/const.py`. (https://github.com/home-assistant/core/blob/5f9a1d105c32fd13d360e53de8e418b13c0c8f56/homeassistant/components/homekit/const.py#L239-L240)

The unit should be ppm (`CONCENTRATION_PARTS_PER_MILLION`) which is already defined.

Icons could be "mdi:molecule-co" and "mdi:molecule-co2"

## Integrations impacted:

**HomeKit**
- [x] https://github.com/home-assistant/core/blob/ac3a6aaa8cdb005b3c900f9d9671e5a859351060/homeassistant/components/homekit/type_sensors.py#L63

**Somfy TaHoma** (CO and CO2)
- [ ] https://github.com/home-assistant/core/pull/46911

**mhz19** (CO2)
- [ ] https://github.com/home-assistant/core/blob/ac3a6aaa8cdb005b3c900f9d9671e5a859351060/homeassistant/components/mhz19/sensor.py

**CO2signal** (CO2)
Unit = CO2eq/{ENERGY_KILO_WATT_HOUR}
- [ ] https://github.com/home-assistant/core/blob/ac3a6aaa8cdb005b3c900f9d9671e5a859351060/homeassistant/components/co2signal/sensor.py

**Awair** (CO2)
- [x] https://github.com/home-assistant/core/blob/ac3a6aaa8cdb005b3c900f9d9671e5a859351060/homeassistant/components/awair/const.py

**foobot** (CO2)
- [ ] https://github.com/home-assistant/core/blob/ac3a6aaa8cdb005b3c900f9d9671e5a859351060/homeassistant/components/foobot/sensor.py#L42

**fibaro** (CO2)
- [x] https://github.com/home-assistant/core/blob/625bbe623812ca2ea6b45371157b9bc2e33727b6/homeassistant/components/fibaro/sensor.py#L30

**tasmota** (CO2)
- [x] https://github.com/home-assistant/core/blob/974e099e2a9527d38445531c6d9bc1461ba4c36f/homeassistant/components/tasmota/sensor.py#L55

**netatmo** (CO2)
- [x] https://github.com/home-assistant/core/blob/f16ccae78bcb7ceecdf8d955f542ec1bf15c140a/homeassistant/components/netatmo/sensor.py#L54

**ambient_station** (CO2)
- [x] https://github.com/home-assistant/core/blob/87795929525b0a89bc316543adbfe83aae48d41c/homeassistant/components/ambient_station/__init__.py#L158

**Comment by Adminiuga** [comment]
There's also Zigbee https://titanproducts.com/zigbee-3-0-co2-sensor/ Zigbee sensors available for CO2 and support for ZHA is coming

**Reply by iMicknl** [reply]
@Adminiuga would you like to give this a try, or should I work on implementing this feature? 👍🏻

**Reply by Adminiuga** [reply]
https://github.com/home-assistant/core/pull/47487

**Reply by iMicknl** [reply]
Added front-end PR: https://github.com/home-assistant/frontend/pull/8602.

**Comment by MartinHjelmare** [comment, marked as answer]
I think we can say that this suggestion has approval and a PR with implementation is welcome. Please see a recent PR that extended device class support for examples.
